### DISCUSSION TITLE
How do I get the currently opened folder?

### DISCUSSION THREAD

**Discussion by dagoston93** [eclipse-theia/theia#13604] [General] [answered]
Hi,
I seem not to be able find a probably very simple API call.
As my IDE starts up, I need to get the currently opened directory's URI.

I created a FrontendApplicationContribution, whose onStart() method get's called properly.
I managed to find out that I can sometimes get the currently open folder from the window.location.hash property, but if the IDE starts up using the last used directory, It doesn't work, as it will be empty.
Also I am unable to use WorkspaceService, as I am not using workspaces, just simple folders...
I tried google, and the discussions on github, but no result.

I guess it's trivial then, but I am overlooking it...

Thanks in advance for any help.
BR,
Agoston

**Comment by msujew** [comment, marked as answer]
Hey @dagoston93,

even just a folder is still a workspace. Therefore, using `WorkspaceService` is still correct. You should be able to use the `roots` property to get all currently open folders (even if it is just one).

**Reply by dagoston93** [reply]
Thank you, I was trying to use getWorkspaceRootUri() and that was the problem then...Now it works like a charm :)
